tests/functional/spec/components/radio.spec.js

```diff
@@ -0,0 +1,236 @@
+const helpers = require('../../helpers');
+const RadioMandatoryPage = require('../../pages/surveys/radio/radio-mandatory.page');
+const RadioNonMandatoryPage = require('../../pages/surveys/radio/radio-non-mandatory.page');
+const SummaryPage = require('../../pages/surveys/radio/radio-summary.page');
+
+describe('Radio button with an option', function() {
+
+var radio_schema = 'test_radio.json';
+
+it('Given an mandatory option is available, when the user clicks the mandatory option then they should be taken to the next page', function() {
+    return helpers.openQuestionnaire(radio_schema).then(() => {
```

Comment:
I think we need to create more structure for these tests, that can be mirrored across the other component tests. We could do this by nesting describe statements e.g. 
```
describe('Component: Radio', function() {
    
    describe('Optional', function() {
        describe('When I submit without choosing an option', function() {
            <actions to do the when go here> 
            it('should go to the next page', function() { ... }
        }
    }
    
    describe('Mandatory', function() {
        describe('When I submit without choosing an option', function() {
            <actions to do the when go here>
            it('should stay on the same page', function() { ... }
            it('should display an error', function() { ... }
        }
    }
}
```
- The first describe for all components would be in the format 'Component: {component name}
- The second level describes would be consistent across all components (except for additional groups of tests specific to a component e.g. the other field on radio/checkboxes)
- We should look at using a third level describe for the when statements